lib/pbench/server/api/resources/upload_api.py

```diff
@@ -48,10 +50,15 @@ def __init__(self, config, logger):
             )
         )
 
+    @Auth.token_auth.login_required()
     def put(self, controller):
 
-        # FIXME: This should be assigned from the decoded authorization token
-        username = "pbench"
+        try:
+            user = User.query(id=Auth.token_auth.current_user().id)
+        except Exception:
+            self.logger.exception("Tarfile upload: Exception querying the user table")
+            abort(500, message="INTERNAL ERROR")
+        username = user.username
```

Comment:
I'm tempted to say we should take this opportunity to switch to registering the dataset under User.id here; however that'll break gold tests, and needs to be accompanied by an equivalent change to `pbench-state-manager`; so maybe it's better to defer that.